src/constraints.jl

```diff
@@ -43,18 +43,44 @@ function addconstraints! end
 # default fallback
 addconstraints!(model::ModelLike, funcs, sets) = addconstraint!.(model, funcs, sets)
 
+
 """
-## Modify Function
+    canset(model::ModelLike, ::ConstraintSet, c::ConstraintIndex{F,S}, ::Type{S})::Bool
 
-    canmodifyconstraint(model::ModelLike, c::ConstraintIndex{F,S}, ::Type{F})::Bool
+Return a `Bool` indicating whether the set in constraint `c` can be replaced by
+another set of the same type `S` as the original set.
+"""
+canset(model::ModelLike, ::ConstraintSet, c::ConstraintIndex, ::Type{S}) where S = false
 
-Return a `Bool` indicating whether the function in constraint `c` can be replaced by another function of the same type `F` as the original function.
+"""
+    set!(model::ModelLike, ::ConstraintSet, c::ConstraintIndex{F,S}, set::S)
+
+Change the set of constraint `c` to the new set `set` which should be of the
+same type as the original set.
+
+### Examples
+
+If `c` is a `ConstraintIndex{F,Interval}`
+
+```julia
+set!(model, ConstraintSet(), c, Interval(0, 5))
+set!(model, ConstraintSet(), c, NonPositives)    # Error
+```
+"""
+function set!(model::ModelLike, cs::ConstraintSet, c::ConstraintIndex{F,S1}, s::S2) where F where  S1 where S2
```

Comment:
The issue of having a specific method like this is that it prevents models to handle it with a more general method as this would lead to ambiguity (this affects mockoptimizer, bridges, ...)
We could instead call something like `seterror` in  line 234.
The default `seterror` method would give the ArgumentError currently at line 234 and we would add this method as a `seterror` method.